src/odemis/acq/move.py

```diff
@@ -0,0 +1,318 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 27 July 2020
+
+@author: Éric Piel, Bassim Lazem
+
+Copyright © 2020 Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms  of the GNU General Public License version 2 as published by the Free
+Software  Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY;  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR  PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+from __future__ import division
+
+import logging
+import math
+import threading
+from asyncio import CancelledError
+from concurrent.futures._base import CANCELLED, RUNNING, FINISHED
+
+import numpy
+import scipy
+
+from odemis import model, util
+from odemis.util import executeAsyncTask
+
+MAX_SUBMOVE_DURATION = 60  # s
+LOADING, IMAGING = 0, 1
+ATOL_LINEAR_POS = 100e-6  # m
+ATOL_ROTATION_POS = 1e-3  # rad
+RTOL_PROGRESS = 0.1
+
+
+def getLoadingProgress(current_pos, start_pos, end_pos):
+    """
+    Computes the position on the path between LOADING and IMAGING.
+    If it’s too far from the line between LOADING → IMAGING, then it’s considered out of the path.
+    :param current_pos: Current position of the stage
+    :param start_pos: Either LOADING or IMAGING position to start the movement from
+    :param end_pos: Either LOADING or IMAGING position to end the movement to
+    :return: Percentage of the progress, None if it's far away from of the path
```

Comment:
Please indicate the types of the arguments. the positions are `dict str->float`. The return is `0<=float<=1, or None`. Also, the return value is not a "percentage" (ie, between 0 and 100), but a "ratio".